src/adapters/reddit.js

```diff
@@ -55,7 +57,32 @@ function formatMessage(txt) {
 
 class Reddit extends Adapter {
 
-  async onDeposit (sourceAccount, amount) {
+  extractTipAmount (tipText) {
+    const matches =  tipText.match(/\+\+\+([\d\.]*)[\s{1}]?XLM/i)
+    if (matches) {
+        return new Big(matches[1])
+    }
+    return undefined
+  }
+
+  extractWithdrawal (body) {
```

Comment:
Is it a good idea to have a hook func for this in the abstract or not? I'm not sure - do u need it on slack as well?

Reply:
Each platform should allow for users to make withdrawals, but each one may be different from the last. I think it makes sense to have extractWithdrawal() as part of some interface, but as per the issue I created states I'm not convinced the abstract-adapter is the place for it. Might make sense to have an extractor class corresponding to each platform as well.